### ISSUE TITLE
Hint text in form labels GOV.UK Frontend review application legacy pages

### ISSUE THREAD

**Issue by nickcolley** [alphagov/govuk-frontend#1407]
This issue is from a May 2019 external accessibility audit report.

**WCAG Reference**: Usability feedback only, there is no WCAG related guidelines.
**Issue ID**: DAC_Issue23
**URL**: http://govuk-frontend-v3.herokuapp.com/full-page-examples/applicant-details

## Summary
Hint text had been provided, however this appeared within the label

## Screen Shot
![](https://user-images.githubusercontent.com/2445413/58713972-a44acf80-83bb-11e9-83d8-300de342e3a4.png)

Hint text had been provided in relation to the ‘Full Name’ form field, however this had been included within the forms label itself which can result in a lot of information reading to the user.

## Current Code Ref(s)
```html
<div class="form-group"> <label class="form-label" for="full-name"> Full name <span class="form-hint"> Please enter your name as it’s written on official documents such as a passport or driving licence. </span> </label> <input class="form-control" id="full-name" type="text" name="full-name" value=""> </div>
```

## Screen reader comments
> “While navigating out of context I found that the hint information relating to the ‘First names’ field appeared while I was situated within the forms dialogue menu and while using the ‘f’ key. This could mean that GDS style guidelines are not met as users should only be able to locate hint information while using the ‘Tab’ key or the arrow keys with desktop screen reading softwares and while swiping through the page with mobile devices. This occurred using JAWS, NVDA and VoiceOver for iPhone and Mac.”

## Solution
Ensure that the <span> element containing the hint text is placed after the form label and contains an ID.
Aria-describedby can be used on the input field to associate it with the hint text.
DAC | Accessibility Report
71
Example: <div class="form-group"> <label class="form-label" for="full-name"> Full name </label> <span class="form-hint" id="full-name-hint"> Please enter your name as it’s written on official documents such as a passport or driving licence. </span> <input class="form-control" id="full-name" type="text" name="full-name" value="" aria-describedby="full-name-hint"> </div>

**Comment by nickcolley** [comment]
This page has examples from GOV.UK Elements so we can test GOV.UK Frontend compatibility with the older library.

So we expect these inputs to fail, and have already improved GOV.UK Frontend to address this.

Based on this I will close this issue out as it's not an issue in the newer GOV.UK Frontend code.
